internal/internal_event_handlers.go

```diff
@@ -818,3 +857,30 @@ func (weh *workflowExecutionEventHandlerImpl) handleChildWorkflowExecutionTermin
 
 	return nil
 }
+
+func (weh *workflowExecutionEventHandlerImpl) handleSignalExternalWorkflowExecutionRequested(event *m.HistoryEvent) error {
+	attributes := event.ExternalWorkflowExecutionSignalRequestedEventAttributes
+	signalID := string(attributes.Control)
+	decision := weh.decisionsHelper.handleSignalExternalWorkflowExecutionRequested(signalID)
+	signal := decision.getData().(*scheduledSignal)
+	if signal.handled {
+		return nil
```

Comment:
What is the situation when signal is already handled?

Reply:
Not sure. I follow pattern from timer. Probably when `handleSignalExternalWorkflowExecutionRequested` or `handleSignalExternalWorkflowExecutionFailed` be called twice.